src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -13,7 +13,7 @@ public class ServiceEndpoint
 
         public virtual string Name { get; internal set; }
 
-        public string Endpoint { get; }
+        public string Endpoint => AccessKey?.Endpoint;
```

Comment:
AccessKey should never be null?

Reply:
I'm afraid we don't prevent this at the compile level.
And I think for a dynamic property, it is much reasonable to acquire a `null` value rather than get a *NullReferenceException* while something unexpected happens.